cupy/core/core.pyx

```diff
@@ -2743,14 +2745,23 @@ not_equal = create_comparison(
 cpdef ndarray _convert_object_with_cuda_array_interface(a):
     cdef Py_ssize_t sh, st
     desc = a.__cuda_array_interface__
+    # TODO(leofang): enforce compliance to the latest protocol?
     shape = desc['shape']
     dtype = numpy.dtype(desc['typestr'])
+    if 'mask' in desc:
+        mask = desc['mask']
+        if mask is not None:
+            raise ValueError("CuPy currently does not support masked arrays.")
+    # TODO(leofang): for ver.2, 'strides' is always available, so don't check.
     if 'strides' in desc:
         strides = desc['strides']
-        nbytes = 0
-        for sh, st in zip(shape, strides):
-            nbytes = max(nbytes, abs(sh * st))
-    else:
+        if strides is not None:
```

Comment:
I think you can unindent this block by 1 level to eliminate the duplicate line `nbytes = internal.prod(shape) * dtype.itemsize`.

Reply:
Good point. I worked in the midnight and couldn't think of a better way! Thanks.